_temp/hud/src/sections/section-ajax.js

```diff
@@ -7,128 +7,148 @@ var process = require('./util/process');
 };
 var recordItem = function(html, selector, stack, length) {
-	//Set row
-	var row = $(html).prependTo(selector);
-	setTimeout(function() {
-		row.addClass('added');
-	}, 1);
-
-	//Track state of the details
-	if (stack.length >= length)
-		stack.shift().remove();
-	stack.push(row);
+    //Set row
+    var row = $(html).prependTo(selector);
+    setTimeout(function() {
+        row.addClass('added');
+    }, 1);
+
+    //Track state of the details
+    if (stack.length >= length)
+        stack.shift().remove();
+    stack.push(row);
 };
 
 var render = function(details, opened) {
-	process.init(structure);
+    process.init(structure);
 
-	return rendering.section(structure, details, opened);
+    return rendering.section(structure, details, opened);
 };
 var postRender = function() {
-	var open = XMLHttpRequest.prototype.open; 
-	XMLHttpRequest.prototype.open = function(method, uri) {
-		if (util.isLocalUri(uri) && uri.indexOf('/glimpse/') == -1) { 
-			this._uri = uri;
-			
-			var startTime = new Date().getTime(); 
-			this.addEventListener('readystatechange', function() {
-					if (this.readyState == 4 && this.getResponseHeader('__glimpse-id'))  { 
-						update(method, uri, new Date().getTime() - startTime, this.getResponseHeader('Content-Length'), this.status, this.statusText, new Date(), this.getResponseHeader('Content-Type'), this.getResponseHeader('__glimpse-id'));
-					}
-				}, false); 
-		}
-
-		open.apply(this, arguments);
-	};    
-	
-	var send = XMLHttpRequest.prototype.send; 
-	XMLHttpRequest.prototype.send = function() {
-		if (this._uri){
-			this.setRequestHeader('__glimpse-isAjax', true);
-		}
-		
-		send.apply(this, arguments);
-	};                      
+    ready = true;
+
+    for (var task in preRenderCache) {
+        task();
+    }
+    preRenderCache = undefined;
 };
 
+function registerListeners() {
+    var open = XMLHttpRequest.prototype.open;
+    XMLHttpRequest.prototype.open = function(method, uri) {
+        if (util.isLocalUri(uri) && uri.indexOf('/glimpse/') == -1) {
+            this._uri = uri;
+
+            var startTime = new Date().getTime();
+            this.addEventListener('readystatechange', function() {
+                    if (this.readyState == 4 && this.getResponseHeader('X-Glimpse-ContextId')) {
+                        if (ready) {
+                            update(method, uri, new Date().getTime() - startTime, this.getResponseHeader('Content-Length'), this.status, this.statusText, new Date(), this.getResponseHeader('Content-Type'), this.getResponseHeader('X-Glimpse-ContextId'));
+                        }
+                        else {
+                            preRenderCache.push(function() {
+                                update(method, uri, new Date().getTime() - startTime, this.getResponseHeader('Content-Length'), this.status, this.statusText, new Date(), this.getResponseHeader('Content-Type'), this.getResponseHeader('_X-Glimpse-ContextId'));
+                            })
```

Comment:
Minor: Can the two calls to `update()` be extracted into a separate, common function?  It has a long list of repeated arguments.